android/KMEA/app/src/main/java/com/tavultesoft/kmea/data/LanguageResource.java

```diff
@@ -45,5 +64,69 @@ public int hashCode() {
 
   public abstract Bundle buildDownloadBundle();
 
-  //public abstract boolean equals();
+  public LanguageResource() {
+    // Noop
+  }
+
+  /**
+   * Constructor using JSON Objects from installed language resource list
+   * @param installedObj
+   */
+  public LanguageResource(JSONObject installedObj) {
```

Comment:
Do we need this constructor? I can't see it called anywhere in `Keyboard` or `LexicalModel` and I don't think it should be, either -- we are relying on the `fromJSON` call to be calling the `super()`

Reply:
good catch. hopefully that's the last nail